packages/plugins/FileService/src/SNSAdaptor/modals/RenameDialog/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { forwardRef, useState } from 'react'
+import type { SingletonModalRefCreator } from '@masknet/shared-base'
+import { useSingletonModal, type InjectedDialogProps } from '@masknet/shared'
+import { RenameDialog } from './RenameDialog.js'
+
+export interface RenameModalOpenProps extends Omit<InjectedDialogProps, 'title' | 'onSubmit' | 'open'> {
+    title?: string
+    currentName?: string
+    message?: React.ReactNode | string
+    description?: React.ReactNode | string
+    onSubmit?(result: string | null): void
+}
+
+export interface RenameModalProps {}
+
+export const RenameModal = forwardRef<SingletonModalRefCreator<RenameModalOpenProps>, RenameModalProps>(
+    (props, ref) => {
+        const [props_, setProps_] = useState<RenameModalOpenProps>()
+        const [open, dispatch] = useSingletonModal(ref, {
```

Comment:
Use `dispatch.close(props)` and the props are the result of `modal.openAndWaitForClose()`.